libraries/commerce/product/subscriptions/index.js

```diff
@@ -47,6 +49,11 @@ function product(subscribe) {
       dispatch(setProductVariantId(null));
     }
   });
+
+  subscribe(productReceived$, ({ dispatch, action }) => {
```

Comment:
The title should only be set if the user is still on the product page with the received id.